docarray/documents/helper.py

```diff
@@ -191,10 +190,10 @@ class MyAudio(NamedTuple):
         assert issubclass(Doc, Audio)
 
     """
-    if __base__:
-        if not issubclass(__base__, BaseDoc):
-            raise ValueError(f'{__base__} is not a BaseDoc or its subclass')
+    if '__base__' in kwargs:
```

Comment:
I don't like removing the __base__ from the arg and put it in the kwargs. Why was it needed? we need to keep it otherwise we lose the information about the type hint of __base__

Reply:
I referred to the create doc from typed dict function and used that. I'll add it back...